File: Hash/930.Binary-Subarrays-With-Sum/930.Binary-Subarrays-With-Sum.py
class Solution:
    def numSubarraysWithSum(self, A, S):
        """
        :type A: List[int]
        :type S: int
        :rtype: int
        """        
        count,result = 0,0
        Map = collections.defaultdict(int)
        Map[0] = 1                
        for i in range(len(A)):
            count+=A[i]
            result+=Map[count-S]
            Map[count]+=1
        return result

File: Hash/930.Binary-Subarrays-With-Sum/930.Binary-Subarrays-With-Sum.cpp
class Solution {
public:
    int numSubarraysWithSum(vector<int>& A, int S) 
    {
        unordered_map<int,int>Map;
        Map[0] = 1;
        int sum = 0, ret = 0;
        for (int i=0; i<A.size(); i++)
        {
            sum += A[i];
            if (Map.find(sum-S)!=Map.end())
                ret+=Map[sum-S];
            Map[sum]+=1;
        }
        return ret;

    }
};
